CONTRIBUTING.md

```diff
@@ -1,112 +1,81 @@
 
-Here is a list of such functions:
- - DataFrame.values
- - `DataFrame.__iter__` and the array protocol `__array__`
 
-Other frameworks like Dask or Molin have a low-level block representation of a multidimensional array that Spark lacks. Until such representation is available, these functions should not be considered.
+### Pandas functions that are NOT considered for inclusion in Koalas
 
-### Spark functions that are temporarily included in Koalas
-
-- Column.alias
-- DataFrame.collect? `.to_pandas` is more complete
-- DataFrame.describe
-- DataFrame.distinct alias for `unique`
-- DataFrame.join alias for `.merge`
-- DataFrame.limit alias for `.head`
-
-### Spark functions that should be included in Koalas
-
-- pyspark.range
+A few categories of functions are not considered for now to be part of the API, for different reasons:
 
-Streaming functions
+1. *Low level and deprecated functions* (Frame, user types, DataFrame.ix)
 
-Functions to control the partitioning, and in-memory representation of the data.
+2. *Functions that have an unexpected performance impact*
 
-Functions that add SQL-like functionalities to DataFrame.
+    These functions (and the caller of theses functions) assume that the data is represented in a compact format (numpy in the case of pandas). Because these functions would force the full collection of the data and because there is a well-documented workaround, it is recommended that  they are not included. 
 
-Functions that present algorithms specific to distributed datasets
-(approx quantiles for example)
+    The workaround is to force the materialization of the pandas DataFrame, either by calling:
+      - `.to_pandas()` (koalas only)
+      - `.to_numpy()` (works with both pandas and koalas)
 
-- DataFrame.approxQuantile, cache, checkpoint, coalesce, colRegex, createGlobalTempView, createOrReplaceGlobalTempView, createOrReplaceTempView, createTempView, crossJoin, crosstab, dropDuplicates / drop_duplicates, explain, hint, intersect, intersectAll, isLocal, isStreaming, localCheckpoint, persist, printSchema, registerTempTable, repartition, repartitionByRange, rollup, schema, select, selectExpr, subtract, take, toDF, unionByName, withWatermark
+    Here is a list of such functions:
+    - DataFrame.values
+    - `DataFrame.__iter__` and the array protocol `__array__`
 
+3. *Low-level multidimensional arrays*: Other frameworks like Dask or Molin have a low-level block representation of a multidimensional array that Spark lacks. Until such representation is available, these functions should not be considered.
 
-### Reading and writing files.
 
-TODO: Koalas methods for reading and writing should work for both local and distributed files.
+### Spark functions that should be included in Koalas
 
+- pyspark.range
```

Comment:
Since they block the access to data fields, I am thinking that all the functions specific to spark (like physical layout, caching, etc.), they should be put under a `spark` accessor: `k_df.spark.cache()` returns a koala df that has been cached. What do you think?

Reply:
we probably need to try a few and see ...